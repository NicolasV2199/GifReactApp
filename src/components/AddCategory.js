import React, { useState } from 'react';
import PropTypes from 'prop-types';



//Recibo por parametros la funcion de set categories para agregar nuevas
export const AddCategory = ({ setCategories }) => {

    //Uso el hook useState para poder cambiar el valor del input
    const [inputValue, setInputValue] = useState('');

    //Me encargo de que cuando haya un cambio en el input, se muestre el nuevo valor
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    /* //Cuando se presione la tecla Enter prevengo que se refresque la pagina, y hago la validacion de que si lo que se inserta es una categoria, entonces llamo a la funcion setCategories que me retorna las categorias que ya tenia, mas el valor insertado en el input, y reinicio el input a vacio*/
    const handleSubmit = (e) =>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories( cat => [ inputValue, ...cat,] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                value={ inputValue } 
                onChange={ handleInputChange } 
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
