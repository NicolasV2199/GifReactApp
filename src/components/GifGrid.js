import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs( category );

    //Se utiliza para que la funcion getgifs solo se llame cuando se renderiza el componente por primera vez
    /* Tiene un parametro que es un arreglo de dependencias, en este caso, vacio */
    //Se agrega la categoria en el arreglo de dependencias por si acaso llegase a cambiar.

    return (
        <>
            <h2 className='animate__animated animate__fadeInDown'> {category} </h2>

            {loading && <p className='animate__animated animate__flash animate__slow'>Loading...</p>}

            <div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
