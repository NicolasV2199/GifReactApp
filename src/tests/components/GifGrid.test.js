import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el GifGrid', () => {
  
    const category = 'Naruto';

    test('Debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
      
        const wrapper = shallow( <GifGrid category= { category } />);

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando carga imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            title: 'Naruto',
            url: 'https://localhost/algo.jpg'  
        }, 
        {
            id: '123',
            title: 'sasuke',
            url: 'https://localhost/algo.jpg'
            
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
      
        const wrapper = shallow( <GifGrid category= { category } />);

        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    

});


