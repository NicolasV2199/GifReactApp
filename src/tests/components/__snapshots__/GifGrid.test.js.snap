// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeInDown"
  >
     
    Naruto
     
  </h2>
  <p
    className="animate__animated animate__flash animate__slow"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
